src/components/Contacts/ContactsTable/ContactsTable.test.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import { GqlMockedProvider } from '../../../../__tests__/util/graphqlMocking';
+import { ContactsQuery } from '../../../../pages/accountLists/[accountListId]/Contacts.generated';
+import { ContactsTable } from './ContactsTable';
+
+const accountListId = '111';
+
+//TODO: Need test coverage for error state
+
+describe('ContactFilters', () => {
+  it('loading', async () => {
+    const { queryByTestId, queryByText } = render(
+      <GqlMockedProvider<ContactsQuery>>
+        <ContactsTable accountListId={accountListId} />
+      </GqlMockedProvider>,
+    );
+
+    expect(queryByText('Loading')).toBeVisible();
+    expect(queryByText('No Data')).toBeNull();
+    expect(queryByText('Error:')).toBeNull();
+    expect(queryByTestId('ContactRows')).toBeNull();
+  });
+
+  it('contacts loaded', async () => {
+    const { queryByTestId, queryByText } = render(
+      <GqlMockedProvider<ContactsQuery>>
+        <ContactsTable accountListId={accountListId} />
+      </GqlMockedProvider>,
+    );
+
+    await waitFor(() => expect(queryByText('Loading')).not.toBeInTheDocument());
+
+    expect(queryByText('Loading')).toBeNull();
+    expect(queryByText('No Data')).toBeNull();
+    expect(queryByText('Error:')).toBeNull();
+    expect(queryByTestId('ContactRows').childNodes.length).toEqual(3);
+  });
+
+  it('empty', async () => {
+    const mocks = {
+      Contacts: {
+        contacts: {
+          nodes: [],
+        },
+      },
+    };
+
+    const { queryByTestId, queryByText } = render(
+      <GqlMockedProvider<ContactsQuery> mocks={mocks}>
```

Comment:
Huh, looks like mocks worked for you here. I was never specifying the top level query name... Looks like you have `Contacts` at the top here. That sounds like my automock bug.

Maybe you've solved it? 🎉 If so it was user error 🤦

Reply:
Yep, this seems to fix it! I just fixed this test: https://github.com/CruGlobal/mpdx-react/blob/811726cf039830f5449da17790f2ec28f6c6aa7b/pages/accountLists/%5BaccountListId%5D/contacts.test.tsx#L38-L40
by adding the top level query name!